### PR TITLE
[Merged by Bors] - feat(Order/WellFounded): well-founded of strictly monotone

### PR DESCRIPTION
**Pull request by apnelson1**
This PR is separated off from #12273, which no longer depends on it. 

We show that an order with a strictly monotone function into an order with WellFoundedLT is itself WellFoundedLT. 
We also add three dual variants, and provide an instance showing that a Nat-graded order is thus well-founded. 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by JADekker** [comment]
Mathematically this is not my expertise, but in terms of style I think this looks good to me!

File: Mathlib/Order/WellFounded.lean [hunk 1]
```diff
@@ -286,3 +286,27 @@ theorem WellFounded.induction_bot {α} {r : α → α → Prop} (hwf : WellFound
 #align well_founded.induction_bot WellFounded.induction_bot
 
 end Induction
+
+section WellFoundedLT
+
+variable [Preorder α] [Preorder β] {f : α → β}
+
+theorem WellFoundedLT.of_strictMono [WellFoundedLT β] (hf : StrictMono f) : WellFoundedLT α where
```
Comment (j-loreaux) [inline Mathlib/Order/WellFounded.lean:294]: I would actually name this different to unlock dot notation. Same for the others in this file.
```suggestion
theorem StrictMono.wellFoundedLT [WellFoundedLT β] (hf : StrictMono f) : WellFoundedLT α where
```

**Comment by j-loreaux** [comment]
Looks fine other than the comment above.

bors d+

**Pull request closed without merging**